rest/route/agent.go

```diff
@@ -153,6 +153,35 @@ func (h *agentCheckGetPullRequestHandler) Parse(ctx context.Context, r *http.Req
 }
 
 func (h *agentCheckGetPullRequestHandler) Run(ctx context.Context) gimlet.Responder {
+	t, err := task.FindOneId(h.taskID)
+	if err != nil {
+		return gimlet.NewJSONInternalErrorResponse(errors.Wrapf(err, "getting task '%s'", h.taskID))
+	}
+	if t == nil {
+		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    fmt.Sprintf("task '%s' not found", h.taskID),
+		})
+	}
+	p, err := patch.FindOne(patch.ByVersion(t.Version))
+	if err != nil {
+		return gimlet.NewJSONInternalErrorResponse(errors.Wrapf(err, "getting patch for task '%s'", h.taskID))
+	}
+	if p == nil {
+		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    fmt.Sprintf("patch for task '%s' not found", h.taskID),
+		})
+	}
+	// This field is only set for PR merge patches.
+	if p.GithubPatchData.MergeCommitSHA != "" {
+		return gimlet.NewJSONResponse(apimodels.PullRequestInfo{
+			// Validation elsewhere in GetMergeablePullRequest ensures that all merge patches are mergeable.
+			Mergeable:      utility.TruePtr(),
+			MergeCommitSHA: p.GithubPatchData.MergeCommitSHA,
+		})
+	}
+
```

Comment:
random comment placement. 
for this: 

> What In think we can actually do is get rid of the agent's /pull_request endpoint altogether since the task config [stores](https://gist.github.com/hadjri/5a52e3f1f04db4982d650b384f585aa1#file-task_config-go-L98) the patch data on it already, but I wanted this to be an easy thing to revert if needed without having to worry about agent rollover issues.

Do we maybe want to add some logging to it and then create a ticket to check a few days later if it logged since it was deployed and if not remove it?

Reply:
Yes, created a ticket for that: https://jira.mongodb.org/browse/EVG-20196